velox/common/file/File.cpp

```diff
@@ -142,21 +148,54 @@ uint64_t LocalReadFile::preadv(
   // Dropped bytes sized so that a typical dropped range of 50K is not
   // too many iovecs.
   static thread_local std::vector<char> droppedBytes(16 * 1024);
+  uint64_t totalBytesRead = 0;
   std::vector<struct iovec> iovecs;
   iovecs.reserve(buffers.size());
+
+  auto readvFunc = [&]() -> ssize_t {
+    const auto bytesRead =
+        folly::preadv(fd_, iovecs.data(), iovecs.size(), offset);
+    if (bytesRead < 0) {
+      LOG(ERROR) << "preadv failed with error: " << folly::errnoStr(errno);
+    } else {
+      totalBytesRead += bytesRead;
+      offset += bytesRead;
+    }
+    iovecs.clear();
+    return bytesRead;
+  };
+
   for (auto& range : buffers) {
     if (!range.data()) {
       auto skipSize = range.size();
       while (skipSize) {
         auto bytes = std::min<size_t>(droppedBytes.size(), skipSize);
+
+        if (iovecs.size() >= IOV_MAX) {
+          ssize_t bytesRead{0};
```

Comment:
You can just put ssize_t bytesRead{0}; right above the for loop and declare there once?